source/agora/node/FullNode.d

```diff
@@ -1005,66 +1005,24 @@ public class FullNode : API
             return;
         }
 
-        UTXO utxo;
-        Output found_output;
-        scope UTXOFinder utxo_finder;
-        scope GetPenaltyDeposit getPenaltyDeposit;
-        if (this.ledger.peekUTXO(enroll.utxo_key, utxo))
+        scope UTXOFinder utxo_finder = (in Hash hash, out UTXO found_utxo) @safe nothrow
         {
-            utxo_finder = (in Hash hash, out UTXO found_utxo)
-            {
-                if (hash == enroll.utxo_key)
-                {
-                    found_utxo = utxo;
-                    return true;
-                }
-                return false;
-            };
-            getPenaltyDeposit = &this.ledger.getPenaltyDeposit;
-        }
-        else
+            return this.ledger.peekUTXO(hash, found_utxo) || this.pool.peekUTXO(hash, found_utxo);
+        };
+        scope GetPenaltyDeposit getPenaltyDeposit = (Hash hash)
         {
-            /// FIXME: Use a proper type and sensible memory allocation pattern
-            // We create a extra UTXO set using the transaction pool if there
-            // is no UTXO in the UTXO set. We only use transactions that have
-            // a frozen UTXO with the right amount.
-            version (all)
-            {
-                foreach (ref Hash hash, ref Transaction tx; this.pool)
-                {
-                    if (!tx.isFreeze())
-                        continue;
-
-                    foreach (size_t idx, output_; tx.outputs)
-                    {
-                        if (output_.type == OutputType.Freeze &&
-                            output_.value >= Amount.MinFreezeAmount)
```

Comment:
What about these checks?

Reply:
enrollment validation routine checks that anyways, so they were redundant